### PR TITLE
Remove link_to_library(goto_functions, symbol_table, ...) [blocks: #4056]

### PR DESCRIPTION
**Pull request by tautschnig**
There are no in-tree uses of this function. External users should use
link_to_library(goto_model, ...). This is in preparation of further changes that
will only support the goto_modelt-variant.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- n/a Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: f20f3072a67bc012ac669df9e56d9a64511dbe3e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/102647097

**Comment by kroening** [comment]
If this doesn't have urgency, then perhaps deprecate that goto_functions/symbol_tablet variant?

**Comment by kroening** [comment]
Or, while at it, all other functions that take the goto_functions/symbol_tablet pair?

**Comment by tautschnig** [comment]
I have created #4303 to do the deprecation instead. There is still a bunch of work left on #4056, and until that's resolved this indeed isn't urgent.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: deacf8b3bf042e5c07ddc83f659b6ab8b3eff4f6).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/103505153

**Comment by tautschnig** [comment]
@kroening This is now ready for review.

**Review by martin-cs — APPROVED** [review]
Seems reasonable as it has been deprecated for 21 months or so.

**Review by smowton — APPROVED** [review]

File: src/goto-programs/link_to_library.h [hunk 1]
```diff
@@ -18,19 +18,11 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <util/deprecate.h>
 #include <util/irep.h>
 
-class goto_functionst;
 class goto_modelt;
 class message_handlert;
 class symbol_tablet;
 
 DEPRECATED(SINCE(2019, 2, 28, "Use link_to_library(goto_model, ...) instead"))
-void link_to_library(
```
Comment (kroening) [inline src/goto-programs/link_to_library.h:27]: Shouldn't that `DEPRECATED` line go away?

Comment (tautschnig) [reply src/goto-programs/link_to_library.h:27]: Thank you for catching this!

**Pull request merged**